app/core/client-page.tsx

```diff
@@ -65,15 +76,15 @@ function createClientPage<TPathParams extends ZodType<any, any>, TQueryParams ex
       if (roles && roles.length > 0) {
         const allowed = arrayIntersection(roles, _roles).length > 0;
         if (!allowed) {
-          return router.push(fallbackUrl);
+          handleAccessRedirect();
```

Comment:
after handle redirect; please `return` in the next line to ensure no any logic keeps running after; and same for other places.